### PR TITLE
Adds Chakra UI components for the Add Item page

### LINKED ISSUE #37: 19. Add Item Design: As a user, I want a Chakra UI drawer component implemented on the List page to hold the Add Item page.
## Summary

The Add Item page will be held in a Chakra UI drawer component that will be accessible through a button on the List page.

## Acceptance Criteria:

- [ ]  A Chakra UI drawer component should be implemented to hold the Add Item page.
- [ ]  The drawer should be accessible from the List page, allowing users to open and close it.
- [ ]  When the drawer is opened, the Add Item page should be displayed within the drawer.

## Notes:

- The Chakra UI drawer component should be implemented with proper styling and positioning to ensure it enhances the overall user experience.
- The Add Item page should be seamlessly integrated into the drawer, maintaining consistent design and functionality.
- Users should be able to easily navigate between the List page and the Add Item page within the drawer.
- Consider the Chakra UI recommendation for [using a form in a drawer](https://chakra-ui.com/docs/components/drawer#using-a-form-in-a-drawer)
- [Figma wireframe](https://www.figma.com/file/4oxU5wehZPNinZpr24UuLw/Chakra-UI-Figma-Kit-(Community)?type=design&node-id=2138-2189&t=cCWXr9PcMt29OwcS-4)

### LINKED ISSUE #38: 18. Add Item Design: As a user, I want to be able to enter item details in the Add Item page within the Chakra UI drawer.
## Summary

The Add Item page should have a consistent and visually appealing user interface by implementing Chakra UI components for the form and input fields.

## Acceptance Criteria:

- [ ]  The Add Item page within the drawer includes fields for entering item details, such as name and the timeframe for buying the item again.
- [ ]  Chakra UI components for input fields should have appropriate styling, including labels, placeholders, and error handling.
- [ ]  The form should have a submit button, styled using Chakra UI, to allow users to submit the item details.
- [ ] The error and added messages should be displayed to the user using the chakra `toast` component.

## Notes:

- Ensure that the Chakra UI components used for forms and input fields adhere to accessibility guidelines.
- Consider utilizing Chakra UI's utility classes and responsive design features to enhance the flexibility and responsiveness of the Add Item page.
- [Figma Wireframe](https://www.figma.com/file/4oxU5wehZPNinZpr24UuLw/Chakra-UI-Figma-Kit-(Community)?type=design&node-id=2138-2276&t=cCWXr9PcMt29OwcS-4)
add-item
- Chakra Toast documentation: https://chakra-ui.com/docs/components/toast

### PR DESCRIPTION
**Pull request by ticiadev**
## Description

This code adds Chakra UI components to the Add Item and List pages. A drawer component has a button on the List page to access the Add Item page, where it has been updated with Chakra UI form and toast components. This simplifies the code by adding Chakra UI components that enhance the design and usability of the page, while also keeping a consistent interface with the rest of the application.

## Related Issue

Closes #38 
Closes #37 

## Acceptance Criteria

- [x]  The Add Item page within the drawer includes fields for entering item details, such as name and the timeframe for buying the item again.
- [x]  Chakra UI components for input fields should have appropriate styling, including labels, placeholders, and error handling.
- [x]  The form should have a submit button, styled using Chakra UI, to allow users to submit the item details.
- [x] The error and added messages should be displayed to the user using the chakra `toast` component.
- [x]  A Chakra UI drawer component should be implemented to hold the Add Item page.
- [x]  The drawer should be accessible from the List page, allowing users to open and close it.
- [x]  When the drawer is opened, the Add Item page should be displayed within the drawer.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |

### Before

image

### After

image

image

## Testing Steps / QA Criteria

In your terminal, use `git checkout jf-tf-add-item-styling` to create a new branch and switch over.
Use `git pull origin jf-tf-add-item-styling` to bring the changes from remote to local.
npm start to start a local instance of our app

- Open the drawer from the List page and confirm that the Add Item page is displayed within it
- Test the drawer accessibility by opening and closing it using both keyboard and mouse
- Add an item to the list and verify that it appears on the list page
- Test error handling by entering invalid data and confirm that appropriate error messages are displayed using the Chakra toast component

### CODE REVIEW

**Review by katherineyuneman — APPROVED** [review]
Everything is looking great even on mobile size and functioning as it should!  I love the way the drawer pops out.

File: src/components/AddItem.jsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+	const onItemChange = (e) => setItemName(e.target.value);
+
+	// function to validate itemName input that returns error messages if any - called onFormSubmit
+	function validateItemInput(data, trimmedItemName) {
+		// 1- checks for empty inputs
+
+		if (!trimmedItemName) {
+			toast({
+				title: 'No item name entered',
```
Comment (Yaosaur) [inline src/components/AddItem.jsx]: @katherineyuneman Same here, do you think having a title is necessary? If so, we should add one to list item page. I feel like takes more space and is basically a repeat of the description.

File: src/components/AddItem.jsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+		if (potentialMatch) {
+			toast({
+				title: 'Item already on list',
+				description: `You are already tracking ${potentialMatch.name}.`,
+				variant: 'errorToast',
+			});
+			return false;
+		}
+		return true;
```
Comment (Yaosaur) [inline src/components/AddItem.jsx]: image

Miso is already on the list. I am expecting to see both `Please only use alphanumeric, ', -, or & characters in your item name.` and  `Miso** is already on your list as Miso.` Would probably be best to have as two separate toasts. In the original implementation we implemented as an array of error messages which was later looped. I think the same pattern can be followed but to return multiple toasts with the error messages.

For reference:
image

File: src/components/AddItem.jsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+				<DrawerOverlay />
+				<DrawerContent bg={'brand.500'} justify="center">
+					<DrawerCloseButton />
+					<DrawerHeader>
+						<Heading>Add Item</Heading>
+					</DrawerHeader>
+					<DrawerBody>
+						<form onSubmit={onFormSubmit}>
+							<Flex direction="column" justify="center" gap={4}>
+								<Text htmlFor="itemName">Item Name:</Text>
```
Comment (Yaosaur) [inline src/components/AddItem.jsx]: Not focusing on pressing the label, try FormLabel instead.

**Review by Yaosaur — CHANGES_REQUESTED** [review]
Works as intended on both desktop and mobile. Couple of changes requested, but otherwise great work!

**Pull request merged**